flink-ml-lib/src/test/java/org/apache/flink/ml/feature/MaxAbsScalerTest.java

```diff
@@ -0,0 +1,271 @@
+    }
+
+    @Test
+    public void testTransformDataWithNullValue() throws Exception {
+        List<Row> predictData =
+                new ArrayList<>(
+                        Arrays.asList(
+                                Row.of(Vectors.dense(0.0, 3.0)),
+                                Row.of((Object) null),
+                                Row.of((Object) null)));
+
+        Table predictDataWithInvalidData =
+                tEnv.fromDataStream(env.fromCollection(predictData)).as("input");
+
+        MaxAbsScaler maxAbsScaler = new MaxAbsScaler();
+
+        MaxAbsScalerModel maxAbsScalerModel = maxAbsScaler.fit(trainDataTable);
+        Table output = maxAbsScalerModel.transform(predictDataWithInvalidData)[0];
+        List<Row> result = IteratorUtils.toList(tEnv.toDataStream(output).executeAndCollect());
+        for (Row row : result) {
+            if (row.getField("output") != null) {
+                double[] expected = new double[] {0.0, 0.0075};
+                DenseVector vec = row.getFieldAs("output");
+                for (int i = 0; i < vec.size(); ++i) {
+                    assertEquals(expected[i], vec.get(i), 1.0e-5);
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testFitAndPredictSparse() throws Exception {
+        MaxAbsScaler maxAbsScaler = new MaxAbsScaler();
+        MaxAbsScalerModel maxAbsScalerModel = maxAbsScaler.fit(trainSparseDataTable);
+        Table output = maxAbsScalerModel.transform(predictSparseDataTable)[0];
+        verifyPredictionResult(output, maxAbsScaler.getOutputCol(), true);
+    }
+
+    @Test
+    public void testSaveLoadAndPredict() throws Exception {
+        MaxAbsScaler maxAbsScaler = new MaxAbsScaler();
+        MaxAbsScaler loadedMaxAbsScaler =
+                TestUtils.saveAndReload(
+                        tEnv, maxAbsScaler, tempFolder.newFolder().getAbsolutePath());
+        MaxAbsScalerModel model = loadedMaxAbsScaler.fit(trainDataTable);
+        MaxAbsScalerModel loadedModel =
+                TestUtils.saveAndReload(tEnv, model, tempFolder.newFolder().getAbsolutePath());
+        Table output = loadedModel.transform(predictDataTable)[0];
+        verifyPredictionResult(output, maxAbsScaler.getOutputCol(), false);
+    }
+
+    @Test
+    public void testGetModelData() throws Exception {
+        MaxAbsScaler maxAbsScaler = new MaxAbsScaler();
+        MaxAbsScalerModel maxAbsScalerModel = maxAbsScaler.fit(trainDataTable);
+        Table modelData = maxAbsScalerModel.getModelData()[0];
+        assertEquals(
+                Collections.singletonList("maxVector"),
+                modelData.getResolvedSchema().getColumnNames());
+        DataStream<Row> output = tEnv.toDataStream(modelData);
+        List<Row> modelRows = IteratorUtils.toList(output.executeAndCollect());
+        assertEquals(new DenseVector(new double[] {200.0, 400.0}), modelRows.get(0).getField(0));
+    }
+
+    @Test
+    public void testSetModelData() throws Exception {
```

Comment:
Could you please add a test case that sets a model data that contains zero and negative values, and make sure the resulting behavior is the same as that in Spark?